### DISCUSSION TITLE
Generate flow docs with lein project

### DISCUSSION THREAD

**Discussion by SuperCuber** [jpmonettas/flow-storm-debugger#62] [Q&A]
Hey, I read this page https://github.com/jpmonettas/flow-storm-debugger/blob/flow-docs/docs/flow_docs.md
and wanted to try it out, but my project is made with leiningen and I don't have `clj` installed, so the sample command doesn't work.
I found this https://jpmonettas.github.io/flow-storm-debugger/user_guide.html#_clojure but I'm not sure how to translate the arguments to something lein would accept.

Could you help me with an equivalent lein command? Also, do you know if it's possible to run this as a lein plugin, without editing `project.clj`?

Thanks :D

**Comment by jpmonettas** [comment]
Hi! Sure, you can run flow-storm.api/cli-doc from your clojure repl also, doesn't matter if you start it with lein or clj.

First you need to add the flow-storm-dbg dependency `[com.github.jpmonettas/flow-storm-dbg "3.3-alpha-290"]` so it is on your classpath,
you can add it to your dev profile, or wherever you put your dev tools.
 
Then to generate your documentation, once you are in your lein repl you can require and call the function : 

```
(require '[flow-storm.api :as fs-api])

(fs-api/cli-doc {:result-name "test-project-flow-docs-1.4.0"
                 :print-unsampled? true
                 :instrument-ns #{"test-project-base-ns"}
                 :fn-symb 'test-project-base-ns.test/test-clj
                 :fn-args []
                 :examples-pprint? true
                 :examples-print-length 2
                 :examples-print-level 3})
```

You can add that to your dev.clj if you have one, or create a script so you have it when you need it. That will generate a jar file, in this case is going to be `test-project-flow-docs-1.4.0.jar`.

Now to browse documentation you need to bring docs jars on your classpath, lein should be able to do this like this : 

```
(defproject test-project "0.1.0-SNAPSHOT"
  ...
  :resource-paths ["resources/test-project-flow-docs-1.4.0.jar"])
```

now you can start your repl and start the debugger like : 

```
(require '[flow-storm.api :as fs-api])
(fs-api/local-connect)
```

And that is it. There are a bunch of steps but you need to do them once (or whenever you want to update the docs)

I hope that works, jump into #flow-storm on clojurians slack if you have any questions.
